pkg/preparer/result_set_test.go

```diff
@@ -0,0 +1,34 @@
+package preparer
+
+import (
+	"testing"
+
+	. "github.com/square/p2/Godeps/_workspace/src/github.com/anthonybishopric/gotcha"
+	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/pods"
+)
+
+func TestZipBasic(t *testing.T) {
+	intent := []kp.ManifestResult{
+		{Manifest: pods.Manifest{Id: "foo"}},
+		{Manifest: pods.Manifest{Id: "bar"}},
+	}
+	reality := []kp.ManifestResult{
+		{Manifest: pods.Manifest{Id: "baz"}},
+		{Manifest: pods.Manifest{Id: "bar"}},
+	}
+
+	ValidatePairList(t, ZipResultSets(intent, reality))
```

Comment:
I'd like this test to be a bit more precise: have the expected outcomes explicitly stated, ie "there's a pair with intent foo, but no reality foo", etc